test/Validate.Layout.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2015 The Khronos Group Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and/or associated documentation files (the
+// "Materials"), to deal in the Materials without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Materials, and to
+// permit persons to whom the Materials are furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Materials.
+//
+// MODIFICATIONS TO THIS FILE MAY MEAN IT NO LONGER ACCURATELY REFLECTS
+// KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
+// SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
+//    https://www.khronos.org/registry/
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+// Validation tests for Logical Layout
+
+#include "gmock/gmock.h"
+#include "UnitSPIRV.h"
+#include "ValidateFixtures.h"
+
+#include <functional>
+#include <sstream>
+#include <string>
+#include <utility>
+
+using std::function;
+using std::ostream_iterator;
+using std::pair;
+using std::stringstream;
+using std::string;
+using std::tie;
+using std::tuple;
+using std::vector;
+
+using ::testing::HasSubstr;
+
+using ValidateLayout = spvtest::ValidateBase<
+    tuple<int, pair<string, function<bool(int)>>>,
+    SPV_VALIDATE_LAYOUT_BIT>;
+
+namespace {
+
+// returns true if order is equal to VAL
+template <int VAL>
+bool Equals(int order) {
+  return order == VAL;
+};
+
+// returns true if order is between MIN and MAX(inclusive)
+template <int MIN, int MAX>
+bool Range(int order) {
+  return order >= MIN && order <= MAX;
+};
+
+// clang-format off
+// SPIRV source used to test the logical layout
+vector<string> instructions = {
```

Comment:
static duration variable is a class.
This I care less about since it's in test code. Similarly, we don't care about performance so you could just make a function that returns this as a newly constructed (not static) value.
I bet you can make this Plain Old Data array and use std::begin and std::end (including std::end(a) - std::begin(a) for the size).